File: apps/backend-go/internal/core/domain/banner/entity/banner.go
package entity

import (
	"errors"
	"time"
)

// Banner represents a banner entity in the domain
type Banner struct {
	ID           string
	ImageTH      string
	ImageEN      string
	URLTH        string
	URLEN        string
	SegmentTiers []string
	StartDate    time.Time
	EndDate      time.Time
	IsActive     bool
	Priority     int
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

// Validate performs business logic validation
func (b *Banner) Validate() error {
	// Date range validation
	if b.EndDate.Before(b.StartDate) || b.EndDate.Equal(b.StartDate) {
		return errors.New("end_date must be after start_date")
	}

	// Priority validation
	if b.Priority < 0 {
		return errors.New("priority must be non-negative")
	}

	return nil
}

File: apps/web/src/types/banner.ts
export interface Banner {
  id: string;
  image_th: string;
  image_en: string;
  url_th: string;
  url_en: string;
  segment_tiers: string[];
  start_date: string; // ISO 8601
  end_date: string;
  is_active: boolean;
  priority: number;
  created_at?: string;
  updated_at?: string;
}

export interface CreateBannerDTO {
  image_th: string;
  image_en: string;
  url_th: string;
  url_en: string;
  segment_tiers: string[];
  start_date: string;
  end_date: string;
  is_active: boolean;
  priority: number;
}

export interface UpdateBannerDTO {
  image_th: string;
  image_en: string;
  url_th: string;
  url_en: string;
  segment_tiers: string[];
  start_date: string;
  end_date: string;
  is_active: boolean;
  priority: number;
}

export interface PriorityUpdate {
  id: string;
  priority: number;
}
